src/jsp/jsp_struct_value.cpp

```diff
@@ -0,0 +1,718 @@
+    return size;
+  }
+
+  void
+  sp_value::pack (cubpacking::packer &serializator) const
+  {
+    if (value == nullptr)
+      {
+	return;
+      }
+
+    pack_value_internal (serializator, *value);
+  }
+
+  size_t
+  sp_value::get_packed_value_size_internal (cubpacking::packer &serializator, std::size_t start_offset, DB_VALUE &v) const
+  {
+    DB_TYPE type = DB_VALUE_TYPE (&v);
+
+    size_t size = start_offset + serializator.get_packed_int_size (start_offset); /* type */
+    switch (type)
+      {
+      case DB_TYPE_INTEGER:
+	size += serializator.get_packed_int_size (size);
+	size += serializator.get_packed_int_size (size);
+	break;
+
+      case DB_TYPE_SHORT:
+	size += serializator.get_packed_int_size (size);
+	size += serializator.get_packed_short_size (size);
+	break;
+
+      case DB_TYPE_BIGINT:
+	size += serializator.get_packed_int_size (size);
+	size += serializator.get_packed_bigint_size (size);
+	break;
+
+      case DB_TYPE_FLOAT:
+	size += serializator.get_packed_int_size (size);
+	size += OR_FLOAT_SIZE;
+
+	// FIXME: no alignment ?
+	// size += DB_ALIGN (size, OR_FLOAT_SIZE) - size + OR_FLOAT_SIZE;
+
+	// TODO: update packer
+	// size += serializator.get_packed_float_size (size);
+	break;
+
+      case DB_TYPE_DOUBLE:
+      case DB_TYPE_MONETARY:
+	size += serializator.get_packed_int_size (size);
+	size += OR_DOUBLE_SIZE;
+
+	// FIXME: no alignment
+	// size += DB_ALIGN (size, OR_DOUBLE_SIZE) - size + OR_DOUBLE_SIZE;
+
+	// assert (false);
+	// TODO: update packer
+	// size += serializator.get_packed_double_size (size);
+	break;
+
+      case DB_TYPE_NUMERIC:
+      {
+	char str_buf[NUMERIC_MAX_STRING_SIZE];
+	numeric_db_value_print (value, str_buf);
+	size += serializator.get_packed_int_size (size); /* dummy length */
+	size += serializator.get_packed_c_string_size (str_buf, strlen (str_buf), size);
+      }
+      break;
+
+      case DB_TYPE_CHAR:
+      case DB_TYPE_NCHAR:
+      case DB_TYPE_VARNCHAR:
+      case DB_TYPE_STRING:
+	size += serializator.get_packed_int_size (size);
+	size += serializator.get_packed_c_string_size (db_get_string (value), db_get_string_size (value), size);
+	break;
+
+      case DB_TYPE_BIT:
+      case DB_TYPE_VARBIT:
+	// NOTE: This type was not implemented at the previous version
+	assert (false);
+	break;
+
+      case DB_TYPE_OBJECT:
+#if !defined (SERVER_MODE)
+	size += serializator.get_packed_int_size (size); /* size */
+	size += serializator.get_packed_oid_size (size);
+#else
+	// TODO: Implement a way to pack DB_TYPE_OBJECT value on Server
+	assert (false);
+#endif
+	break;
+
+      case DB_TYPE_DATE:
+      case DB_TYPE_TIME:
+	size += serializator.get_packed_int_size (size);
+	size += serializator.get_packed_int_size (size);
+	size += serializator.get_packed_int_size (size);
+	size += serializator.get_packed_int_size (size);
+	break;
+
+      case DB_TYPE_TIMESTAMP:
+	size += serializator.get_packed_int_size (size);
```

Comment:
If they are arranged in the same type, direct calculation is possible as shown below.
size += (OR_INT_SIZE * 6);
If the same type, not variable length, is used repeatedly, only the first of them is moved by align, and after that it seems to just fit.

Reply:
add comments for readability
ex) size += serializator.get_packed_int_size (size); // year

Reply:
@ctshim 
I'd like to keep this pattern as it is. because of the following reasons
1) consistency how to use this serializer
2) To not care about how the aligning is done
3) There is no particular impact on performance.

@byungwook-kim 
Good point. I'm going to add comments

Reply:
OK, Thank you for your kindness.